src/irclib.py

```diff
@@ -1913,6 +1916,22 @@ def _maybeStartSasl(self, msg):
                         if x.lower() in available]
             self.tryNextSaslMechanism(msg)
 
+    def startSasl(self, msg):
```

Comment:
You should be able to dedup this function with `_maybeStartSasl` above

Reply:
I *think* we should unconditionally call resetSasl() when starting a new SASL transaction as I've done here. But I'm not quite certain as this function kind of mixes configuration and the sasl state machine state initialization. Is this an issue? Do I need to split those in half?

Reply:
seems fine